src/components/TodoItem/index.jsx

```diff
@@ -1,14 +1,33 @@
 import React from 'react';
-
 import styles from './styles.module.sass';
-
-function TodoItem() {
+import {CompleteButton, DeleteButton, EditButton} from "../common/Button";
+import {useDispatch} from "react-redux";
+import {completeTODO, removeTodo} from "../../redux/action-creators";
+import {useHistory} from 'react-router-dom'
+function TodoItem({todo}) {
+  const dispatch = useDispatch();
+  const history = useHistory();
+  // add completed class if the todo is completed
+  const completedClass = todo.completed ? styles.completed : '' ;
+  function deleteTodo() {
+    dispatch(removeTodo(todo.id));
+  }
+  function completeTodo() {
+    dispatch(completeTODO(todo.id));
+  }
+  function editHandler() {
+    history.push(`/edit/${todo.id}`)
+  }
   return (
-    <div className={styles.card}>
-      <h2 className={styles.title}>Title</h2>
-      <button className={styles.completeBtn}>Complete</button>
-      <button className={styles.deleteBtn}>Delete</button>
-    </div>
+    <article className={styles.card}>
+      <div className={styles.textContainer}>
+        <h2 className={`${styles.title} ${completedClass}`}>{todo.text}</h2>
+        <p className={styles.time}>{todo.time}</p>
+      </div>
+      {!todo.completed &&  <CompleteButton onClick={completeTodo}>Complete</CompleteButton>}
```

Comment:
You could have added the ability to undo the complete action using the same button. Now the user have to edit the whole todo just to maek it as 'In Progress'